### PR TITLE
Fix set boolean properties on form field

### PR DESCRIPTION
**Pull request by kuzmany**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Steps to test this PR:

Properties for bolean fields are set as `yes => Yes` to properties. It make issue for some cases. This PR fixed it.

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create form with email and boolean and checkbox group with boolean contact field mapping and enabled **Use assigned contact/company field's list choices.**
image
3. Submit form with Yes option
4. Go to campaign and create campaign with `form field condition` for checkboxgroup field  with preselected option Yes
5. Run campaign. 
6. See like condition go to false path, even we expect true
7. Condition compare `'yes' === true`, after PR should go to true

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11171"></a>

### CODE REVIEW

**Comment by escopecz** [comment]
@kuzmany I'm removing some PRs that are marked with the `WIP` label out of the planned release. I can't tell whether the PR is ready to test but based on the WIP label it looks like it's not.

**Comment by escopecz** [comment]
I went through the code and I think the code change should be safe. It is used only to generate form choices in the FormBundle. However, when I was testing this I did not find any difference in behavior from the 4.x branch. Here is what I did:
1. Created 2 boolean custom fields: "Are you Happy?" and "Really?"
2. Created form 
3. Added a radio group button linked to bool field  "Are you Happy?" using options from the contact field
4. Added a radio checkbox button linked to bool field "Really?" using options from the contact field
5. Added an email field
6. Saved the form
7. Created segment for all contacts having an email address
8. Created campaign
9. Added form field condition "Are you NOT happy?" with "Are you happy" field = NO
10. Added form field condition "Really?" with "Really?" field = YES
11. Added 4 "Add tag" actions to each end of the conditions to know which will execute.

Then I submitted several contacts through the form and let them run through the campaign. Here is the result metrics:
- Contact with Happy: empty, Really: empty got yesHappy and notReally tags ⛔ 
- Contact with Happy: yes, Really: no got yesHappy and notReally tags ✅ 
- Contact with Happy: no, Really: yes got yesHappy and notReally tags ⛔ 
- Contact with Happy: yes, Really: yes got yesHappy and notReally tags ⛔ 

So basically the condition the same no matter what is the value and no matter whether I use your fix or not.

Screen Shot 2022-06-17 at 17 51 07


Screen Shot 2022-06-17 at 18 03 59

**Review by volha-pivavarchyk — APPROVED** [review]
Works the way as described

Form:
image

Campaign:
image

Result on the current branch:
image

Result on the 4.x branch:
image

**Review by biozshock — APPROVED** [review]
Repeated what @escopecz did. For me it works as expected.

**Comment by kuzmany** [comment]
Close favor to https://github.com/mautic/mautic/pull/12866
